src/integrationTest/java/com/mongodb/spark/sql/connector/read/MongoBatchTest.java

```diff
@@ -463,6 +469,66 @@ void testCustomMongoClientFactory() {
     assertTrue(CustomMongoClientFactory.CALLED.get());
   }
 
+  @Test
+  void testReadFromTimeseriesDatabase() {
+    assumeTrue(isAtLeastFiveDotZero() && !isSharded());
```

Comment:
What is the relationship between sharded and partitioners? That is, why is sharded excluded? (Just curious.)

Reply:
No relationship, its just that timeseries sharded support was added in 5.1 and that was causing issues.